addon-test-support/@ember/test-helpers/dom/fill-in.ts

```diff
@@ -43,13 +43,17 @@ export default function fillIn(target: Target, text: string): Promise<void> {
     }
 
     __focus__(element);
-
     if (isControl) {
-      element.value = text;
+      // ref: https://html.spec.whatwg.org/multipage/input.html#concept-input-apply
+      const maxlengthApplicableTypes = ['text', 'search', 'url', 'tel', 'email', 'password'];
+      const { type, tagName } = element;
+      const maxlength = element.getAttribute('maxlength');
+      const shouldValidateMaxlength =
+        maxlength && (maxlengthApplicableTypes.includes(type) || tagName === 'TEXTAREA');
```

Comment:
Also, can you swap this around _slightly_ to something like:

```js
const shouldValidateMaxLength = element.hasAttribute(‘maxlength’) && ((tagName === ‘INPUT’ && maxlengthApplicableTypes.indexOf(type) > -1) || tagName === ‘TEXTAREA’);

element.value = shouldValidateMaxlength ? text.slice(0, maxlength) : text;
```

The main things I’m concerned with here are:

* base shouldValidateMaxlength on element.hasAttribute
* only validate `type` for `input`s

Reply:
Makes sense. I guess this should probably move into a utils/ fn, as typeIn will need need same logic